templates/buildkite-elastic.yml

```diff
@@ -334,20 +340,23 @@ Resources:
                   BOOTSTRAP_SCRIPT="buildkite-agent bootstrap"
                 fi;
 
-                cat << EOF > /etc/buildkite-agent/buildkite-agent.cfg
-                name="$(AWS::StackName)-\$INSTANCE_ID"
-                token="$(BuildkiteAgentToken)"
-                meta-data=\$(printf 'queue=%s,docker=%s,stack=%s,buildkite-aws-stack' "$(BuildkiteQueue)" "\$DOCKER_VERSION" "$(AWS::StackName)")
-                meta-data-ec2=true
-                bootstrap-script="\$BOOTSTRAP_SCRIPT"
-                hooks-path=/etc/buildkite-agent/hooks
-                build-path=/var/lib/buildkite-agent/builds
-                plugins-path=/var/lib/buildkite-agent/plugins
-                EOF
+                for kite in {1..$(BuildkiteAgentsPerInstance)}; do
+                  cat << EOF > /etc/buildkite-agent/buildkite-agent-${kite}.cfg
```

Comment:
we can actually keep the config out of this loop, and use `name="$(AWS::StackName)-%n-\$INSTANCE_ID"` and the buildkite server will give it a numeric number. No need to change the builds directory either, because the agent name given by the server is used for the checkout directory /within/ the builds directory.